src/core/domain.py

```diff
@@ -124,23 +126,29 @@ class DatabaseConnectionInfo:
     dbname: str
 
 
-class ServiceAccountInfo:
-    """Class representing service account and namespace to be used by Kyuubi.
+class SparkServiceAccount(RelationState):
```

Comment:
nit: I still think `ServiceAccountInfo` or `SparkServiceAccountInfo` is a better name (since it is consistent with other state classes like `DatabaseConnectionInfo` and `S3ConnectionInfo`. It's just my preference though.

Reply:
That's all right. I see your argument to keep it consistent. I would honestly actually remove "Info" across the board, as I don't think it is too informative TBH. But let's keep it for the time being across all of them. If this will be hurting the ODC of others (apart from myself), we can remove it later in the future.

Reply:
Done in [cab2310](https://github.com/canonical/kyuubi-k8s-operator/pull/21/commits/cab23106143e22a664b8e50d8b28204a1982bc26)